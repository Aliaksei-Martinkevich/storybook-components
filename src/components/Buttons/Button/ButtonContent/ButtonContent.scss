.button-content {
    display: flex;
    justify-content: center;
    align-items: center;
    flex-flow: row nowrap;
}

.button-content__icon {
    flex: none;
    height: 18px;
    margin-right: 10px;
}

.butten-content__label {
}
