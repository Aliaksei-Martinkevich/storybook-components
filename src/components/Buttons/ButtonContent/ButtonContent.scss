.button-content {
    display: flex;
    justify-content: center;
    align-items: center;
    flex-flow: row nowrap;
}

.button-content__icon {
    flex: none;
    height: 17px;
    margin-right: 10px;
}

.butten-content__label {
}
