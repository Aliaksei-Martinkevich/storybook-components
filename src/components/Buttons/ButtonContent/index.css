.button-content {
    height: 100%;
    display: flex;
}

.button-content:last-child {
    margin-left: 3%;
}

