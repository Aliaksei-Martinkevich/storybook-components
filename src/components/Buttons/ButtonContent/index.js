import React from 'react';
import PropTypes from 'prop-types';
import './index.css';
import LoadIcon from '../LoadIcon';
import ButtonLabel from '../ButtonLabel';

const ButtonContent = ({text, iconUri}) => (
  <div className={'butten-content'}>
    { iconUri
      ? <LoadIcon iconUri={iconUri} />
      : null
    }
    <ButtonLabel text={text} />
  </div>
);

ButtonContent.propTypes = {
  iconUri: PropTypes.string,
  text: PropTypes.string.isRequired,
}

export default ButtonContent;
