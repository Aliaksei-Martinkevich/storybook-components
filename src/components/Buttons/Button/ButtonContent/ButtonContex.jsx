import React from 'react';
import PropTypes from 'prop-types';
import './ButtonContent.scss';
import LoadIcon from './LoadIcon/LoadIcon.jsx';
import ButtonLabel from './ButtonLabel/ButtonLabel.jsx';

const ButtonContent = ({ text, iconUri }) => (
  <div className="button-content">
    { iconUri
      ? <LoadIcon iconUri={iconUri} className="button-content__icon" />
      : null
    }
    <ButtonLabel text={text} className="butten-content__label" />
  </div>
);

ButtonContent.defaultProps = {
  iconUri: '',
};

ButtonContent.propTypes = {
  iconUri: PropTypes.string,
  text: PropTypes.string.isRequired,
};

export default ButtonContent;
