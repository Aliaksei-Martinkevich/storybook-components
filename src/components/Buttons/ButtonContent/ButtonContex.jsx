import React from 'react';
import PropTypes from 'prop-types';
import './ButtonContent.scss';
import LoadIcon from '../LoadIcon/LoadIcon.jsx';
import ButtonLabel from '../ButtonLabel/ButtonLabel.jsx';

const ButtonContent = ({ text, iconUri = '' }) => (
  <div className={'button-content'}>
    { iconUri
      ? <LoadIcon iconUri={iconUri} />
      : null
    }
    <ButtonLabel text={text} />
  </div>
);

ButtonContent.propTypes = {
  iconUri: PropTypes.string,
  text: PropTypes.string.isRequired,
};

export default ButtonContent;
