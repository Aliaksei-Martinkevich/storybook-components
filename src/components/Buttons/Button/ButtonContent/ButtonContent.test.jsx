import React from 'react';
import renderer from 'react-test-renderer';
import ButtonContent from './ButtonContex.jsx';
import icon from '../../../../../assets/icons/svg/loading-circle-dashes-black.svg';

it('Should render without fails', () => {
  const customClassName = 'class-name-example';
  const text = 'Some text';
  const component = renderer.create(
    <ButtonContent className={customClassName} text={text} iconUri={icon} />,
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
