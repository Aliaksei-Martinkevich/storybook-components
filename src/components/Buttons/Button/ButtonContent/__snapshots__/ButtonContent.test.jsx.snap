// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button content Should render without fails 1`] = `
<div
  className="button-content"
>
  <img
    alt=""
    className="load-icon button-content__icon"
    src="test-file-stub"
  />
  <span
    className="button-label butten-content__label"
  >
    Some text
  </span>
</div>
`;
